### DISCUSSION TITLE
Does Python have an interface that can be used for consumers with priorities?

### DISCUSSION THREAD

**Discussion by llx-08** [rabbitmq/rabbitmq-server#12260] [Other]
### Is your feature request related to a problem? Please describe.

Hi, I read the tutorial documents and found that there is a feature for priority-based consumers. However, the provided examples are only in Java. Does the Python version also have this interface?

### Describe the solution you'd like

Create python priority cusumers interface or add it to tutorial, thank you!

**Comment by michaelklishin** [comment]
@llx-08 should we guess what tutorial you have read? I'm not aware of any tutorials that use consumer priorities. It is not a feature you need most of the time.

There is a [dedicated doc guide](https://www.rabbitmq.com/docs/consumer-priority) that explains that you need to set an optional consumer argument, `x-priority`, when registering a consumer.

You can do that with Pika, and there's a [Pika example that uses this specific key](https://github.com/pika/pika/blob/96a92379346285a8f53dd0b3f76cb26f45baf350/examples/async_as_callback.py#L107) when declaring a queue. Except that consumer priority will be passed to [`basic_consume`](https://github.com/pika/pika/blob/96a92379346285a8f53dd0b3f76cb26f45baf350/pika/adapters/blocking_connection.py#L1598-L1604) and not `queue_declare`.
